custom_components/virtual_step_dimmer/light.py

```
@@ -0,0 +1,155 @@
+                "switch",
+                "toggle",
+                {"entity_id": self._switch_entity},
+                blocking=True,
+            )
+            # Add a small delay between toggles to allow the physical device to respond
+            await asyncio.sleep(0.5)
+
+        # We don't set the brightness here, we wait for the sensor to update
+        # and the state listener will update the brightness.
+
+    async def async_turn_off(self, **kwargs: dict[str, Any]) -> None:
```

Ruff (0.14.3)

Unused method argument: `kwargs`

(ARG002)